core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2426,4 +2421,28 @@ public String toString() {
                 ")";
         }
     }
+
+    /**
+     * FetchOffsetMetadata class is used to cache offset and its log metadata.
+     */
+    static final class OffsetMetadata {
+        private long offset;
+        private LogOffsetMetadata offsetMetadata;
+
+        OffsetMetadata() {
+        }
```

Comment:
Should we initialize the fields?

Reply:
I am not sure if initializing the fields has any benefits. The only value initialization inside the constructor that makes sense is `offset=-1` i.e. an invalid case. We don't really need this initialization since we can use `offsetMetadata() == null` to check if the object `fetchOffsetMetadata` has been cached yet or not.

Reply:
Yes, since 0 is a valid offset, it's better to initialize offset to -1.

Reply:
i've pushed the changes in my latest commit. Thanks!